packages/mapviewer/src/modules/menu/components/advancedTools/ImportCatalogue/ProviderList.vue

```diff
@@ -1,49 +1,221 @@
 
 import TextSearchMarker from '@/utils/components/TextSearchMarker.vue'
 
-const { showProviders, providers } = defineProps({
+const { showProviders, groupedProviders, filterApplied } = defineProps({
     showProviders: {
         type: Boolean,
         default: false,
     },
-    providers: {
-        type: Array /* Array of Provider */,
+    groupedProviders: {
+        type: Object /* Object of grouped providers by base URL */,
         default() {
-            return []
+            return {}
         },
     },
+    filterApplied: {
+        type: Boolean,
+        default: false,
+    },
 })
 
+// reactive data
 const emit = defineEmits(['chooseProvider', 'hide'])
-
 const providerList = useTemplateRef('providerList')
+const maxTabIndex = ref(0)
+const treeData = reactive([])
+
+// Function to get the longest common prefix of an array of URLs must end with '/'
+function getLongestCommonPrefix(urls) {
+    if (!urls.length) return ''
+    let prefix = urls[0]
+    for (const url of urls) {
+        while (!url.startsWith(prefix)) {
+            prefix = prefix.slice(0, -1)
+            if (!prefix) break
+        }
+    }
+    // Ensure the prefix ends with a '/'
+    return prefix.endsWith('/') ? prefix : prefix.slice(0, prefix.lastIndexOf('/') + 1)
+}
+
+// Function to build the tree node structure
+// This function takes a base URL and an array of providers and returns a tree node
+// structure. If there is only one provider, it returns a URL node. If there are multiple
+// providers, it groups them by their common prefix and creates a group node with
+// sub-group nodes for each unique prefix.
+//  It also set the name as the text to be shown in the UI
+function buildTreeNode(baseUrl, providers) {
+    if (providers.length === 1) {
+        return {
+            id: baseUrl,
+            name: providers[0].htmlDisplay,
+            type: 'url',
+            url: providers[0].url,
+            emphasize: providers[0].emphasize,
+        }
+    }
+
+    const urls = providers.map((provider) => provider.url)
+    const commonPrefix = getLongestCommonPrefix(urls)
+    const subGroups = {}
+
+    providers.forEach((provider) => {
+        const relativeUrl = provider.htmlDisplay.replace(commonPrefix || baseUrl, '')
+        const subGroupKey = relativeUrl.split('/')[0]
+
+        if (!subGroups[subGroupKey]) {
+            subGroups[subGroupKey] = []
+        }
+        subGroups[subGroupKey].push({
+            ...provider,
+            relativeUrl: relativeUrl.replace(subGroupKey + '/', ''),
+        })
+    })
+
+    return {
+        id: baseUrl,
+        name: commonPrefix || baseUrl,
+        type: 'group',
+        expanded: filterApplied,
+        children: Object.entries(subGroups).map(([key, subGroupProviders]) => {
+            if (subGroupProviders.length === 1) {
+                return {
+                    id: subGroupProviders[0].url,
+                    name: key + '/' + subGroupProviders[0].relativeUrl,
```

Comment:
might make sense to use some `...${variable}...` notation here instead of concat multiple strings